pandas/core/indexes/multi.py

```diff
@@ -2618,8 +2620,9 @@ def _wrap_joined_index(self, joined, other):
     @Appender(Index.isin.__doc__)
     def isin(self, values, level=None):
         if level is None:
-            return algos.isin(self.values,
-                              MultiIndex.from_tuples(values).values)
+            values = MultiIndex.from_tuples(values,
```

Comment:
you *might* be able to change this to ``_ensure_index(values)`` I think (as its more idiomatic). I also don't think you will need the ``.values``